department-of-veterans-affairs/caseflow
# Ki/APPEALS-42573 - Correspondence and Appeal Tasks: Add tasks to linked appeals: Add task modal

Resolves [APPEALS-42573 Correspondence and Appeal Tasks: Inbound Ops Team Superuser/visor: Add tasks to linked appeals: Add task modal](https://jira.devops.va.gov/browse/APPEALS-42573)

# Description
This adds the functionality to add tasks to linked appeals on the Correspondence Details page.

## Acceptance Criteria

## Testing Plan
1. Log in as a Superuser/Supervisor.
2. Build (from Existing Appeals - the page needs to be saved to properly link them) or Intake (from Intake workflow) a Correspondence with linked appeals.
3. If there are less than 4 tasks on the appeal, a button that says "+ Add Task" should populate.
4. On clicking it, a Modal should pop up
5. Fill in the Task and Instructions
6. "Submit" -- The AC calls out that it should say "Next" to prepare for the AutoText portion, but it just submits with that next button right now.
7. Task should populate on the Correspondence Details page with success banner.

Notes:
1. Unsaved, _temporary_ linked appeals do not have this functionality.
2. Non-super users should be able to see the tasks, but not add them with the button.
3. The front-end takes a moment to catch up with the updates, but it should only be a second or two seconds.
4. The modal has submission safeties in place, and requires both the task to be selected, and the text box filled out.


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
